src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/ImageController.php

```diff
@@ -66,10 +67,13 @@ class ImageController extends FrameworkBundleAdminController
      *
      * @return JsonResponse
      */
-    public function getImagesAction(int $productId, Request $request): JsonResponse
+    public function getImagesAction(int $productId, ?int $shopId): JsonResponse
```

Comment:
Sorry, it as not in the PR original scope, but I think we should follow the same convention that was used on all endpoints for the product list. Meaning split into dedicated actions with each one its route something like:
- `admin_products_v2_product_shop_images_for_all_shops` with no parameter
- `admin_products_v2_product_shop_images_for_shop` with required shopId parameter

One of the advantages I noticed with this is that if the shop context is not correctly handled (for any reason) and we don't have a valid shopId it will fail very early (when generating the URL), so it makes the whole feature stricter but safer and more stable